### PR TITLE
Update INetChannel

### PR DESCRIPTION
**Pull request by vanz696**

### CODE REVIEW

File: public/inetchannel.h [hunk 1]
```diff
@@ -57,82 +55,93 @@ enum NetChannelBufType_t : int8
...
+	
 	virtual void	StartRegisteringMessageHandlers( void ) = 0;
 	virtual void	FinishRegisteringMessageHandlers( void ) = 0;
-	virtual void	RegisterNetMessageHandlerAbstract( CUtlSlot *, const CUtlAbstractDelegate &, int, NetMessageHandle_t , int ) = 0;
-	virtual void	UnregisterNetMessageHandlerAbstract( CUtlSlot *, const CUtlAbstractDelegate &, NetMessageHandle_t ) = 0;
+	
+	virtual void	RegisterNetMessageHandlerAbstract( CUtlSlot *nSlot, const CUtlAbstractDelegate &delegate, int nParamCount, INetworkSerializable *pNetMessage, int nPriority ) = 0;
+	virtual void	UnregisterNetMessageHandlerAbstract( CUtlSlot *nSlot, const CUtlAbstractDelegate &delegate, INetworkSerializable *pNetMessage ) = 0;
+	
 	virtual void	SetChallengeNr( unsigned int challenge ) = 0;
```
Comment (GAMMACASE) [inline public/inetchannel.h:117]: In the ``INetworkGameServer`` challenge was changed to a string, can you verify what this method is actually using?

Comment (vanz696) [reply public/inetchannel.h:117]: It looks like it is being passed as an integer

```CNetChan::SetChallengeNr```:
img1

```CNetworkGameServerBase::ConnectClient```:
img2

I assume that it takes this value from C2S_CONNECT
https://github.com/SteamDatabase/GameTracking-CS2/blob/master/Protobufs/connectionless_netmessages.proto#L6

Comment (GAMMACASE) [reply public/inetchannel.h:117]: Hm, interesting, then all good

**Comment by GAMMACASE** [comment]
Thanks as usual!

**Pull request merged**